### PR TITLE
Implement add note endpoint

### PR DESCRIPTION
**Pull request by LBHmsaitz**
Implemented an endpoint that adds notes to Universal Housing objects. It makes use of a stored procedure existent in dev and live environment for adding the notes and even though this supports adding notes to different UH objects (tenancy, repairs, properties etc.) the endpoint currently accepts adding notes to work orders.
The stored procedure used does not return the Id of the added note so the endpoint does return a 204 response without content. 
I have explored a way to retrieve the added note using a second query after executing the stored procedure though I have left it out for now - the current use of the endpoint does not require this and it can be extended in the future if needed.

### CODE REVIEW

File: HackneyRepairs/Tests/Validations/noteRequestValidationTests.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+﻿using System;
```
Comment (zuren) [inline HackneyRepairs/Tests/Validations/noteRequestValidationTests.cs]: Should this file be `NoteRequestValidationTests` ?

File: HackneyRepairs/Validators/NoteRequestValidator.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        public NoteRequest NoteRequest { get; set; }
+
+        public NoteRequestValidationResult(NoteRequest request)
+        {
+            ErrorMessages = new List<string>();
+            Valid = true;
+            NoteRequest = request;
+        }
+    }
+}
```
Comment (zuren) [inline HackneyRepairs/Validators/NoteRequestValidator.cs]: Love the validation object!

File: HackneyRepairs/Repository/UhwRepository.cs [hunk 3]
```diff
@@ -157,6 +158,38 @@ INNER JOIN
...
+                throw new UhwRepositoryException();
+            }
+
+            var parameters = new DynamicParameters();
+            parameters.Add("@KeyObject", note.ObjectKey);
+            parameters.Add("@NoteType", "GLO_GEN");
+            parameters.Add("@SecCat", "002");
+            parameters.Add("@NoteText", note.Text);
+            parameters.Add("@UserId", Environment.GetEnvironmentVariable("DefaultAddNoteUser"));
+            parameters.Add("@ObjectReference", note.ObjectReference);
```
Comment (zuren) [inline HackneyRepairs/Repository/UhwRepository.cs]: Fair few magic values here - wonder if it'd be more readable/reusable if we defined them as constants? e.g.

```
parameters.Add("@NoteType", DEFAULT_NOTE_TYPE)
```

File: HackneyRepairs/HackneyRepairs.csproj [hunk 4]
```diff
@@ -40,6 +40,7 @@
     <PackageReference Include="Microsoft.AspNet.WebApi.Client" Version="5.2.6" />
     <PackageReference Include="Pomelo.EntityFrameworkCore.MySql" Version="2.1.1" />
     <PackageReference Include="NLog.Targets.Syslog" Version="5.1.0" />
+    <PackageReference Include="Bogus" Version="24.3.1" />
```
Comment (zuren) [inline HackneyRepairs/HackneyRepairs.csproj]: Like the use of faker :thu

File: HackneyRepairs/Repository/UhwRepository.cs (hunk 3 above)
Comment (LBHmsaitz) [reply HackneyRepairs/Repository/UhwRepository.cs]: Addressed

**Pull request merged**